File: backend/internal/models/requisition.go
package models

import "time"

type Requisition struct {
	ID              int       `json:"id"`
	RequesterID     int       `json:"requester_id"`
	VendorID        *int      `json:"vendor_id"` // Pointer to allow null
	ItemDescription string    `json:"item_description" validate:"required"`
	Quantity        int       `json:"quantity" validate:"required,gt=0"`
	EstimatedPrice  float64   `json:"estimated_price" validate:"required,gt=0"`
	TotalPrice      float64   `json:"total_price"`
	Justification   string    `json:"justification"`
	Status          string    `json:"status"`
	CreatedAt       time.Time `json:"created_at"`
}

type CreateRequisitionPayload struct {
	VendorID        *int    `json:"vendor_id"`
	ItemDescription string  `json:"item_description" validate:"required"`
	Quantity        int     `json:"quantity" validate:"required,gt=0"`
	EstimatedPrice  float64 `json:"estimated_price" validate:"required,gt=0"`
	Justification   string  `json:"justification"`
}

File: frontend/lib/models/requisition.dart
import 'package:json_annotation/json_annotation.dart';

part 'requisition.g.dart';

@JsonSerializable()
class Requisition {
  final int id;
  @JsonKey(name: 'requester_id')
  final int requesterId;
  @JsonKey(name: 'vendor_id')
  final int? vendorId;
  @JsonKey(name: 'item_description')
  final String itemDescription;
  final int quantity;
  @JsonKey(name: 'estimated_price')
  final double estimatedPrice;
  @JsonKey(name: 'total_price')
  final double totalPrice;
  final String? justification;
  final String status;
  @JsonKey(name: 'created_at')
  final DateTime createdAt;

  Requisition({
    required this.id,
    required this.requesterId,
    this.vendorId,
    required this.itemDescription,
    required this.quantity,
    required this.estimatedPrice,
    required this.totalPrice,
    this.justification,
    required this.status,
    required this.createdAt,
  });

  factory Requisition.fromJson(Map<String, dynamic> json) => _$RequisitionFromJson(json);
  Map<String, dynamic> toJson() => _$RequisitionToJson(this);
}
